server/aesdsocket.c

```diff
@@ -0,0 +1,393 @@
+static int socket_fd = 0;
+static int file_fd = 0;
+static int connection_fd = 0;
+
+/* Function Prototypes */
+static int start_daemon(void);
+static void close_app(void);
+void signal_handler(int signo);
+
+/* Function definitions */
+/**
+ * @brief Starts Daemon by creating a child process
+ *
+ * @param void
+ *
+ * @return int
+ */
+static int start_daemon(void)
+{
+    fflush(stdout);
+    pid_t pid = fork();
+    if (FAILURE == pid)
+    {
+        syslog(LOG_PERROR, "creating a child process:%s\n", strerror(errno));
+        return FAILURE;
+    }
+    else if (0 == pid)
+    {
+        printf("Child process created successfully\n");
+        umask(0);
+        pid_t id = setsid();
+        if (FAILURE == id)
+        {
+            syslog(LOG_PERROR, "setsid:%s\n", strerror(errno));
+            return FAILURE; 
+        }
+        /* change current directory to root */
+        if (FAILURE == chdir("/"))
+        {
+            syslog(LOG_PERROR, "chdir:%s\n", strerror(errno));
+            return FAILURE;       
+        }
+        /* close standard files of process */
+        close(STDIN_FILENO);
+        close(STDOUT_FILENO);
+        close(STDERR_FILENO);
+        
+        /* redirect standard files to /dev/null */
+        int fd = open("/dev/null", O_RDWR);
+        if (FAILURE == fd)
+        {
+            syslog(LOG_PERROR, "open:%s\n", strerror(errno));
+            return FAILURE;       
+        }
+        if (FAILURE == dup2(fd, STDIN_FILENO))
+        {
+            syslog(LOG_PERROR, "dup2:%s\n", strerror(errno));
+            return FAILURE;    
+        }
+        if (FAILURE == dup2(fd, STDOUT_FILENO))
+        {
+            syslog(LOG_PERROR, "dup2:%s\n", strerror(errno));
+            return FAILURE;    
+        }
+        if (FAILURE == dup2(fd, STDERR_FILENO))
+        {
+            syslog(LOG_PERROR, "dup2:%s\n", strerror(errno));
+            return FAILURE;    
+        }
+        close(fd);
+    }
+    else
+    {
+        syslog(LOG_PERROR, "Terminating Parent process");
+        exit(0);
+    }
+    return SUCCESS;
+}
+
+/**
+ * @brief Performs closing steps of the application
+ *
+ * @param void
+ *
+ * @return void
+ */
+static void close_app(void)
+{
+    close(file_fd);
+    close(connection_fd);
+    /* deletes the file */
+    if (FAILURE == unlink(FILENAME))
+    {
+       syslog(LOG_PERROR, "unlink %s: %s", FILENAME, strerror(errno));
+    }
+    close(socket_fd);
+    if (FAILURE == shutdown(socket_fd, 2))
+    {
+        syslog(LOG_PERROR, "shutdown: %s", strerror(errno));
+    }
+
+    /* close the connection to the syslog utility */
+    closelog();
+}
+
+/**
+ * @brief Handles signals SIGINT and SIGTERM
+ *
+ * @param int signo - number of signal received
+ *
+ * @return void
+ */
+void signal_handler(int signo)
+{
+    if ((SIGINT == signo) || (SIGTERM == signo))
+    {
+        syslog(LOG_DEBUG, "Caught signal:%d, exiting", signo);
+        exit_condition = 1;
+        close_app();
```

Comment:
@chchbuff consider handling this cleanup in your main loop after `exit_condition` instead of relying on your signal handler.